src/test/test_proj_data_info_subsets.cxx

```diff
@@ -0,0 +1,220 @@
+{
+}
+
+shared_ptr<ProjData> TestProjDataInfoSubsets::construct_test_proj_data()
+{
+  shared_ptr<Scanner> scanner_ptr(new Scanner(Scanner::E953));
+  shared_ptr<ProjDataInfo>
+    proj_data_info_sptr(ProjDataInfo::construct_proj_data_info(scanner_ptr,
+                                                               /*span*/5, scanner_ptr->get_num_rings()-1,
+                                                               /*views*/ scanner_ptr->get_num_detectors_per_ring()/2/8, 
+                                                               /*tang_pos*/64, 
+                                                               /*arc_corrected*/ false));
+  shared_ptr<ExamInfo> exam_info_sptr(new ExamInfo);
+  return std::make_shared<ProjDataInMemory>(exam_info_sptr, proj_data_info_sptr);
+}
+    
+void
+TestProjDataInfoSubsets::
+run_tests()
+{
+    try
+      {
+        // Open sinogram
+        if (_sinogram_filename.empty())
+          _input_sino_sptr = construct_test_proj_data();
+        else
+          _input_sino_sptr = ProjData::read_from_file(_sinogram_filename);
+
+        test_split(*_input_sino_sptr);
+        // test_split_and_combine(*_input_sino_sptr);
+
+        // shared_ptr<ProjectorByBinPairUsingProjMatrixByBin> proj_pair (new ProjectorByBinPairUsingProjMatrixByBin);
+
+        // shared_ptr<VoxelsOnCartesianGrid<float> > backproj_sptr =
+        //     MAKE_SHARED<VoxelsOnCartesianGrid<float> >(*_input_sino_sptr->get_proj_data_info_sptr());
+        // backproj_sptr ->fill(0.f);
+
+        // proj_pair->set_up(_input_sino_sptr->get_proj_data_info_sptr(), backproj_sptr);
+        // shared_ptr<ForwardProjectorByBin> fwd_proj_sptr = proj_pair->get_forward_projector_sptr();
+        // shared_ptr<BackProjectorByBin> bck_proj_sptr = proj_pair->get_back_projector_sptr();
+
+        // bck_proj_sptr->back_project(*bck_proj_sptr, *fwd_proj_sptr, 0, 1);
+
+        // test_back_projection_is_consistent(*_input_sino_sptr, *backproj_sptr);
+        // test_forward_projection_is_consistent(*_input_sino_sptr, *backproj_sptr);
+    }
+    catch(const std::exception &error) {
+        std::cerr << "\nHere's the error:\n\t" << error.what() << "\n\n";
+        everything_ok = false;
+    }
+    catch(...) {
+        everything_ok = false;
+    }
+}
+
+
+void TestProjDataInfoSubsets::
+test_split(const ProjData &proj_data)
+{
+  int view;    
+  int num_subsets = 4;
+
+  std::vector<int> subset_views;
+
+  for (int subset = 0; subset < num_subsets; ++subset){
+    // create a vector containg every num_subsest-th view starting at subset
+    // for num_subsets = 4 and subset = 0 this is [0, 4, 8, 12, ...]
+    // for num_subsets = 4 and subset = 1 this is [1, 5, 9, 13, ...]
+
+    subset_views.clear();
+    view = subset;
+
+    while(view < proj_data.get_num_views()){
+      subset_views.push_back(view);
+      view += num_subsets;
+    }
+
+    ProjDataInMemory& subset_proj_data = (ProjDataInMemory&)*proj_data.get_subset(subset_views);
```

Comment:
Dangerous and will fail for reasons:
- C-style cast (equivalent to `static_cast`, and the latter is more verbose but much preferred) will not do any run-time checks, so as soon a `get_subset()` would return something else, this would crash with a seg-fault.
- you have a reference to a temp-object, which could/disappear immediately.

As we don't need to know what type it is, we can resolve this either like this
```suggestion
    auto subset_proj_data_uptr = proj_data.get_subset(subset_views);
    auto& subset_proj_data = *auto subset_proj_data_uptr;
```
or just use the first line and replace `proj_data.get*` with `proj_data_uptr->get*` (which I think is cleaner)